src/Lib.P2P/Cid.cs

```diff
@@ -0,0 +1,584 @@
+        /// </summary>
+        /// <param name="stream">
+        ///   The <see cref="CodedInputStream"/> to read from.
+        /// </param>
+        /// <returns>
+        ///   A new <see cref="Cid"/>.
+        /// </returns>
+        public static Cid Read(CodedInputStream stream)
+        {
+            var cid = new Cid();
+            var length = stream.ReadLength();
+            if (length == 34)
+            {
+                cid.Version = 0;
+            }
+            else
+            {
+                cid.Version = stream.ReadInt32();
+                cid.ContentType = stream.ReadMultiCodec().Name;
+            }
+
+            cid.Hash = new MultiHash(stream);
+
+            return cid;
+        }
+
+        /// <summary>
+        ///   Writes the binary representation of the CID to the specified <see cref="CodedOutputStream"/>.
+        /// </summary>
+        /// <param name="stream">
+        ///   The <see cref="CodedOutputStream"/> to write to.
+        /// </param>
+        public void Write(CodedOutputStream stream)
+        {
+            using (var ms = new MemoryStream())
+            {
+                if (Version != 0)
+                {
+                    ms.WriteVarint(Version);
+                    ms.WriteMultiCodec(ContentType);
+                }
+
+                Hash.Write(ms);
+
+                var bytes = ms.ToArray();
+                stream.WriteLength(bytes.Length);
+                stream.WriteSomeBytes(bytes);
+            }
+        }
+
+        /// <summary>
+        ///   Reads the binary representation of the CID from the specified byte array.
+        /// </summary>
+        /// <param name="buffer">
+        ///   The souce of a CID.
+        /// </param>
+        /// <returns>
+        ///   A new <see cref="Cid"/>.
+        /// </returns>
+        /// <remarks>
+        ///   The buffer does NOT start with a varint length prefix.
+        /// </remarks>
+        public static Cid Read(byte[] buffer)
+        {
+            var cid = new Cid();
+            if (buffer.Length == 34)
+            {
+                cid.Version = 0;
+                cid.Hash = new MultiHash(buffer);
+                return cid;
+            }
+
+            using (var ms = new MemoryStream(buffer, false))
+            {
+                cid.Version = ms.ReadVarint32();
+                cid.ContentType = ms.ReadMultiCodec().Name;
+                cid.Hash = new MultiHash(ms);
+                return cid;
+            }
+        }
+
+        /// <summary>
+        ///   Returns the binary representation of the CID as a byte array.
+        /// </summary>
+        /// <returns>
+        ///   A new buffer containing the CID.
+        /// </returns>
+        /// <remarks>
+        ///   The buffer does NOT start with a varint length prefix.
+        /// </remarks>
+        public byte[] ToArray()
+        {
+            if (Version == 0) return Hash.ToArray();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)